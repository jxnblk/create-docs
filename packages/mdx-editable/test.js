import React from 'react'
import { create as render } from 'react-test-renderer'
import {
  code,
  pre,
  LivePreview,
  LiveEditor
} from './src'

const renderJSON = el => render(el).toJSON()

describe('mdx-editable', () => {
  test('LiveEditor renders', () => {
    const json = renderJSON(<LiveEditor code='<h1>hi</h1>' />)
    expect(json).toMatchSnapshot()
  })

  test('LivePreview renders', () => {
    const json = renderJSON(<LivePreview code='<h1>hi</h1>' />)
    expect(json).toMatchSnapshot()
  })

  test('code renders', () => {
    const json = renderJSON(React.createElement(code, {
      className: 'language-.jsx'
    }, 'hi'))
    expect(json).toMatchSnapshot()
  })
})
