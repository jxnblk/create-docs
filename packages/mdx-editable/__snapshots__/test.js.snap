// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mdx-editable LivePreview renders 1`] = `
<div
  className="sc-bdVaJa zQNlg"
>
  <div
    className="react-live"
  >
    <div
      className="react-live-preview"
    >
      <h1>
        hi
      </h1>
    </div>
  </div>
</div>
`;
